technic/machines/register/generator.lua

```diff
@@ -174,7 +187,7 @@ function technic.register_generator(data)
 						}
 					)..pipeworks.button_label
 			end
-			meta:set_string("formspec", generator_formspec..form_buttons)
+			update_generator_formspec(meta, desc, percent, form_buttons)
```

Comment:
I guess percent could be 0 here too, it might be more correct to calculate it here but I think just having 0 here wont really change anything if 0 is same as invisible overlay.
```suggestion
			update_generator_formspec(meta, desc, 0, form_buttons)
```

Reply:
Either these suggestion or make it `nil`. Currently luacheck linter tests fail because `percent` isn't defined at all for these two.